### DISCUSSION TITLE
Is there a way to SuppressKeyPress?

### DISCUSSION THREAD

**Discussion by tkefauver** [TolikPylypchuk/SharpHook#15] [Q&A] [answered]
Hi! First just wanted to say thank you for this very handy library which has been working great for me :)

But I am trying to handle hotkey sequences in my program and in the past have used `e.SuppressKeyPress` (in winforms) when the user is inputting a possible hotkey sequence but I haven't found a method to do so with SharpHook. 

Any idea how I could handle this platform independently?

**Comment by TolikPylypchuk** [comment, marked as answer]
Hi! Thanks for asking!

Yes, suppressing input events is possible, but I guess it's not quite clear. The `HookEventArgs` class contains the `Reserved` property which you can use to suppress an event:

```
e.Reserved = EventReservedValueMask.SuppressEvent;
```

In the next version I'll probably introduce a new property like `SuppressEvent` or something like that so it'll be more clear.

Bear in mind that suppressing events only works on Windows and macOS - it doesn't work on Linux. Also, it must be done in a synchronous event handler, so you need to use `SimpleGlobalHook` instead of `TaskPoolGlobalHook`. If you want to know why then you can read more [here](https://github.com/TolikPylypchuk/SharpHook/issues/8).

**Comment by TolikPylypchuk** [comment]
FYI, in version 4.0.0 the way to suppress events became a little more clear - `HookEventArgs` now contains the `SuppressEvent` property instead of `Reserved`. You can now do this to suppress events:

```c#
e.SuppressEvent = true;
```
